lib/java_buildpack/jre/ibm_jre_initializer.rb

```diff
@@ -0,0 +1,228 @@
+        @droplet.java_home.version = @version
+        super
+      end
+
+      # (see JavaBuildpack::Component::BaseComponent#compile)
+      def compile
+        download(@version, @uri['uri'], @component_name) do |file|
+          check_sha256(file, @uri['sha256sum'])
+          with_timing "Installing #{@component_name} to #{@droplet.sandbox.relative_path_from(@droplet.root)}" do
+            install_bin(@droplet.sandbox, file)
+          end
+        end
+        @droplet.copy_resources
+      end
+
+      # (see JavaBuildpack::Component::BaseComponent#release)
+      def release
+        @droplet
+          .java_opts
+          .add_system_property('java.io.tmpdir', '$TMPDIR')
+        @droplet.java_opts << '-Xtune:virtualized'
+        @droplet.java_opts.concat mem_opts
+        @droplet.java_opts.concat tls_opts
+        @droplet.java_opts << '-Xshareclasses:none'
+      end
+
+      private
+
+      # constant HEAP_RATIO is the ratio of memory assigned to the heap
+      # as against the container total and is set using -Xmx.
+      HEAP_RATIO = 0.75
+
+      KILO = 1024
+
+      # Installs the Downloaded InstallAnywhere (tm) BIN file to the target directory
+      #
+      # @param [String] target_directory, Where the java needs to be installed
+      # @param [File] file, InstallAnywhere (tm) BIN file
+      # @return [Void]
+      def install_bin(target_directory, file)
+        FileUtils.mkdir_p target_directory
+        response_file = Tempfile.new('response.properties')
+        response_file.puts('INSTALLER_UI=silent')
+        response_file.puts('LICENSE_ACCEPTED=TRUE')
+        response_file.puts("USER_INSTALL_DIR=#{target_directory}")
+        response_file.close
+
+        File.chmod(0o755, file.path) unless File.executable?(file.path)
+        shell "#{file.path} -i silent -f #{response_file.path} 2>&1"
+      end
+
+      # Checks the SHA256 Checksum of the file
+      #
+      # @param [File] file, The downloaded file
+      # @param [String] checksum, The string containing the SHA256 of the file
+      def check_sha256(file, checksum)
+        raise 'sha256 checksum does not match' unless Digest::SHA256.hexdigest(File.read(file.path)) == checksum
+      end
+
+      # Returns the max heap size ('-Xmx') value
+      def mem_opts
+        mopts = []
+        total_memory = memory_limit_finder
+        if total_memory.nil?
+          # if no memory option has been set by cloudfoundry, we just assume defaults
+        else
+          calculated_heap_ratio = heap_ratio_verification(heap_ratio)
+          heap_size = heap_size_calculator(total_memory, calculated_heap_ratio)
+          mopts.push "-Xmx#{heap_size}"
+        end
+        mopts
+      end
+
+      # Returns the heap_ratio attribute in config file (if specified) or the HEAP_RATIO constant value
+      def heap_ratio
+        @configuration['heap_ratio'] || HEAP_RATIO
+      end
+
+      def tls_opts
```

Comment:
I'm a bit skeptical of this setting being the default.  We're seeing more and more customers attempting to lock down the versions of TLS supported (all the way down to locking it to TLS v1.2 _only_).  At the very least, forcing a modification of the defaults seems a bit out of bounds.  A user can always `cf set-env <APP> JAVA_OPTS '-Dcom.ibm.jsse2.overrideDefaultTLS=true'` if they would prefer this behavior.